### PR TITLE
Avoid possible length extension attack.

### PR DESCRIPTION
**Pull request by RoSk0**
See https://www.drupal.org/project/drupalauth4ssp/issues/3020308

### CODE REVIEW

File: lib/Auth/Source/External.php [hunk 1]
```diff
@@ -122,14 +123,17 @@ private function getUser()
...
 
             // make sure the hash matches
             // make sure the UID is passed
-            if ((isset($hash) && !empty($hash)) && (isset($uid) && !empty($uid))) {
+            if ((isset($cookie_hash) && !empty($cookie_hash)) && (isset($uid) && !empty($uid))) {
                 // Make sure no one manipulated the hash or the uid in the cookie before we trust the uid
-                $cookie_salt = $this->config->getCookieSalt();
-                if (sha1($cookie_salt . $uid) !== $hash) {
+                $hash = Crypt::hmacBase64(
+                    $account->id(),
```
Comment (plach79) [inline lib/Auth/Source/External.php]: `$account` is not defined

File: lib/Auth/Source/External.php [hunk 2]
```diff
@@ -122,14 +123,17 @@ private function getUser()
...
+                    $this->config->getCookieSalt() . \Drupal::service('private_key')->get()
```
Comment (plach79) [inline lib/Auth/Source/External.php]: Drupal is not bootstrapped yet, so the service cannot be retrieved.

**Comment by plach79** [comment]
With the fixes above and the companion patch for the Drupal module, this seems to be working well.

**Comment by RoSk0** [comment]
@plach79 thanks, your fix incorporate in the branch.

**Pull request merged**